superbench/benchmarks/micro_benchmarks/gpu_copy_performance/gpu_copy.cu

```diff
@@ -426,43 +484,92 @@ int RunCopy(const BenchArgs &args, const Buffers &buffers) {
     }
 
     // Create stream to launch kernels
+    if (SetGpu(args.worker_gpu_id)) {
+        return -1;
+    }
     cuda_err = cudaStreamCreateWithFlags(&stream, cudaStreamNonBlocking);
     if (cuda_err != cudaSuccess) {
         fprintf(stderr, "RunCopy::cudaStreamCreate error: %d\n", cuda_err);
         return -1;
     }
+    // P2P bidirecitonal test, create a second stream
+    if (args.is_bidirectional) {
+        if (SetGpu(args.worker_gpu_id_2)) {
+            return -1;
+        }
+        cuda_err = cudaStreamCreateWithFlags(&stream_2, cudaStreamNonBlocking);
+        if (cuda_err != cudaSuccess) {
+            fprintf(stderr, "RunCopy::cudaStreamCreate error: %d\n", cuda_err);
+            return -1;
+        }
+    }
 
     // Launch jobs and collect running time
     auto start = std::chrono::steady_clock::now();
     for (int i = 0; i < args.num_loops; i++) {
         if (args.is_sm_copy) {
+            if (SetGpu(args.worker_gpu_id)) {
+                return -1;
+            }
```

Comment:
Why SetGpu again and again

Reply:
In bidirectional P2P test, SMCopyKernel or cudaMemcpy is sent to different GPUs in a round robin manner. According to [CUDA stream document](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#stream-and-event-behavior) we have to set to the device before operating a stream on it. Also according do [cudaSetDevice document](https://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__DEVICE.html#group__CUDART__DEVICE_1g159587909ffa0791bbe4b40187a4c6bb) it's a call with very low overhead.

Reply:
Got it. Thanks.